server/src/main/java/org/elasticsearch/rest/RestUtils.java

```diff
@@ -35,6 +36,13 @@ public class RestUtils {
 
     public static final UnaryOperator<String> REST_DECODER = RestUtils::decodeComponent;
 
+    public static void decodeQueryString(URI uri, Map<String, String> params) {
+        final var query = uri.getQuery();
+        if (query != null) {
+            decodeQueryString(query, 0, params);
+        }
+    }
```

Comment:
We call `decodeQueryString(HttpExchange#getRequestURI()#getQuery()` a lot, but `URI#getQuery()` returns null if there's no `?...` in the URL.

> Returns: The decoded query component of this URI, or null if the query is undefined

Rather than requiring an additional null check for every caller, I thought an overload might work here.